### DISCUSSION TITLE
How to load index/data from another milvus instance

### DISCUSSION THREAD

**Discussion by zch-cc** [milvus-io/milvus#25425] [Q&A and General discussion]
Hi community, first thanks for building this cool product. How to load index/data from another milvus instance? Right now I am using standalone(docker) deployment and I have index/data on one machine. How do I upload the data/index to s3 so that when I create a new instance on another machine, I do not need to rebuild the index and reinsert the data and can just download or load from s3?

**Comment by xiaofan-luan** [comment]
this is what you may want tool, https://github.com/zilliztech/milvus-backup, feel free to ask for help

**Reply by zch-cc** [reply]
Thanks for letting me know. I wonder if it is possible to upload certain file in the volumes directory and next time when I start the new instance, I can just copy the directory into the new one so I do not need to do the reinsert step? Want to ask this because the resinsert also takes time and would like a solution for just download the data and directly loading it into the db. Thanks.

**Reply by xiaofan-luan** [reply]
check out the bulk insert features and that's what you are looking for

**Reply by zch-cc** [reply]
Cool. So the procedure is that I create the backup and then upload the backup directory to s3. When I create a new instance, download the backup directory to s3 and use the restore function to restore the dataset?

**Reply by yhmo** [reply]
The milvus-backup tool does these things:
1. Backup data from a milvus cluster A. The tool calls a milvus interface to get a files list of a collection, and download the files from MinIO/S3 to the local path.
2. Restore data to another Milvus cluster B. Firstly, the tool calls Milvus interface to create a new collection with the same schema. Secondly, the tool uploads the data files to MinIO/S3 of cluster B. And calls bulkinsert() interface to restore the data into cluster B without insert(). Just note, the bulkinsert interface is a kind of "back door" to read data files into Milvus storage. It is not direct copying, it also read/parse the data file. Since it doesn't depend on the insert channel, it is faster than insert().


It is not possible to "copy the file to S3 and direct use it", because Milvus is a complicated system, any data consumed into Milvus needs to be processed carefully and precisely.
